online_meetings/bluejeans_report.py

```diff
@@ -33,141 +36,175 @@
-    {\"type\":\"date\",\"comparison\":\"gt\",\"value\":\"" + start.astimezone().isoformat() + "\",\"field\":\"start_time\"}]\
-    &fileName=meetings_30th-Mar-2020_30th-Mar-2020&userid=" + str(ENV.get("BLUEJEANS_USER_ID")) + "&app_name=command_center"
+# output settings
+CREATE_CSVS = ENV.get('CREATE_CSVS', False)
+INVENTORY_DB = ENV['INVENTORY_DB']
+logger.info(INVENTORY_DB)
+APPEND_TABLE_NAMES = ENV.get('APPEND_TABLE_NAMES', ['job_run', 'data_source_status'])
+
+def run_bluejeans_report() -> Sequence[Dict[str, Union[str, pd.Timestamp]]]:
+    logger.info("run_bluejeans_report")
+
+    # Record data source info for Canvas API
+    bluejeans_data_source = {
+        'data_source_name': 'BLUEJEANS_API',
+        'data_updated_at': pd.to_datetime(time.time(), unit='s', utc=True)
+    }
+
+    url = "https://api.bluejeans.com/oauth2/token?Client"
+
+    payload = "{ \"grant_type\": \"client_credentials\", \"client_id\": \"Metrics_BJN\", \"client_secret\": \"" + ENV.get('BLUEJEANS_CLIENT_SECRET') + "\"}"
+    logger.info(payload)
+    headers = {
+    'Accept': 'application/json',
+    'Content-Type': 'application/json'
+    }
+
+    token_expiration_time = datetime.now()
+    response = requests.request("POST", url, headers=headers, data=payload).text.encode('utf8')
+    token_json = json.loads(response)
+    token = token_json["access_token"]
+    logger.info(token)
+    ## the expiration time in seconds
+    expires_in_sec = token_json["expires_in"]
+    ## minus for one minute, for safty check
+    expires_in_sec = expires_in_sec - 60
+    token_expiration_time = datetime.now() + timedelta(seconds=expires_in_sec)
+    logger.info(f"{token} expires on: {token_expiration_time}")
 
     payload = {}
     headers = {
-        'Accept': 'application/json',
-        'Content-Type': 'application/json',
-        'Authorization': f'Bearer {token}'
+    'Accept': 'application/json',
+    'Content-Type': 'application/json',
+    'Authorization': f'Bearer {token}'
     }
-    reportJobs_response = requests.request("POST", url, headers=headers, data=payload)
-    reportJob_json = json.loads(reportJobs_response.text.encode('utf8'))
-    if ("status" in reportJob_json) and (reportJob_json["status"] == "success") and ("jobId" in reportJob_json):
-        # get the job id
-        job_id = reportJob_json["jobId"]
-        logger.info("report job id = " + job_id)
-
-    # sleep to wait for report job finish
-    download_file_path = ""
-    while not download_file_path:
-        # retrieve report job information including the file download path
-        jobs_url = f"https://indigo-api.bluejeans.com/v1/enterprise/{ENV.get('BLUEJEANS_ENTERPRISE_ID')}/indigo/jobs/list?userid={ENV.get('BLUEJEANS_USER_ID')}"
+
+    total_df = pd.DataFrame()
+
+    # TODO: the earliest time is hardcoded now
+    from_date = dateparser.parse(ENV.get('ZOOM_EARLIEST_FROM', '2020-03-31')).date()
+    to_date = dateparser.parse(ENV.get('ZOOM_EARLIEST_TO', '2020-04-09')).date()
```

Comment:
This feels if these environment variables are used for Bluejean and dates align with ZOOM these names can be refactored as  `ONLINE_EARLIEST_FROM`  `ONLINE_EARLIEST_TO`. Also, please update these variables in `env_blank.json`